### PR TITLE
Add tests for stringification of Complex with Rational values

### PR DESCRIPTION
**Pull request by herwinw**

### CODE REVIEW

**Comment by herwinw** [comment]
This is probably more generic in practice: I guess it calls either `#to_s` or `#inspect` on both parts. It might be worth testing this with a mock object instead.

File: core/complex/inspect_spec.rb [hunk 1]
```diff
@@ -13,4 +13,11 @@
     Complex(-1, -4).inspect.should == "(-1-4i)"
     Complex(-7, -6.7).inspect.should == "(-7-6.7i)"
   end
+
+  it "uses parentheses for Rationals" do
```
Comment (eregon) [inline core/complex/inspect_spec.rb:17]: This is confusing because the usage of parentheses is regardless of rationals or not.

Comment (eregon) [reply core/complex/inspect_spec.rb:17]: Ah you mean the inner parentheses.
Yeah that's just Rational# to_s vs inspect.
Indeed it'd be much better to test that with a mock than depending on the details of some other method.

Comment (herwinw) [reply core/complex/inspect_spec.rb]: Well, it took some tries to actually create a Complex with the mocks, but I guess the spec is now correct.
The addition of `*` based on the last character of the stringification of the imaginary part has been based on the C implementation of MRI.

**Review by eregon — APPROVED** [review]

**Pull request merged**